modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/test/java/com/liferay/dynamic/data/mapping/form/field/type/internal/redirect/button/RedirectButtonDDMFormFieldTemplateContextContributorTest.java

```diff
@@ -0,0 +1,215 @@
+
+import com.liferay.dynamic.data.mapping.render.DDMFormFieldRenderingContext;
+import com.liferay.dynamic.data.mapping.test.util.DDMFormTestUtil;
+import com.liferay.dynamic.data.mapping.test.util.DDMFormValuesTestUtil;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactory;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactoryUtil;
+import com.liferay.portal.kernel.test.portlet.MockLiferayPortletURL;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.ResourceBundle;
+
+import javax.portlet.PortletURL;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.mockito.Matchers;
+
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+/**
+ * @author Rodrigo Paulino
+ */
+@PrepareForTest(
+	{
+		LanguageUtil.class, RequestBackedPortletURLFactoryUtil.class,
+		ResourceBundleUtil.class
+	}
+)
+@RunWith(PowerMockRunner.class)
+public class RedirectButtonDDMFormFieldTemplateContextContributorTest {
+
+	@Before
+	public void setUp() throws Exception {
+		_setUpRequestBackedPortletURLFactoryUtil();
+		_setUpResourceBundleUtil();
+	}
+
+	@Test
+	public void testGetParameters1() {
+		_mockLanguageUtilFormat(
+			"message1", new Object[] {"messageArgument1", "messageArgument2"});
+
+		Map<String, Object> parameters =
+			_redirectButtonDDMFormFieldTemplateContextContributor.getParameters(
+				DDMFormTestUtil.createRedirectButtonDDMFormField(
+					new Object[] {
+						DDMFormValuesTestUtil.createLocalizedValue(
+							"buttonLabel", LocaleUtil.US)
+					},
+					new Object[] {"message1"},
+					new Object[] {"messageArgument1", "messageArgument2"},
+					new Object[] {"mvcRenderCommandName"},
+					StringUtil.randomString(),
+					new Object[] {"parameterName=parameterValue"},
+					new Object[] {_PORTLET_ID},
+					new Object[] {
+						DDMFormValuesTestUtil.createLocalizedValue(
+							"title", LocaleUtil.US)
+					}),
+				_mockDDMFormFieldRenderingContext());
+
+		Assert.assertEquals("buttonLabel", parameters.get("buttonLabel"));
+		Assert.assertEquals(
+			"messageArgument1,messageArgument2,message1",
+			parameters.get("message"));
+		Assert.assertEquals(
+			"http//localhost/test?portletId_mvcRenderCommandName=" +
+				"mvcRenderCommandName;portletId_parameterName=parameterValue",
+			parameters.get("redirectURL"));
+		Assert.assertEquals("title", parameters.get("title"));
+	}
+
+	@Test
+	public void testGetParameters2() {
```

Comment:
@rodrigopaulino, please rename to `testGetParametersWithoutMessageArgumentsAndParameters`.

Reply:
Done.